### ISSUE TITLE
The motion of the CharacterBody2D will get stuck with static objects

### ISSUE THREAD

**Issue by CodeLazier** [appsinacup/godot-rapier-physics#191]
Non-linear motion, transverse elliptical shapes are stuck in rectangular shapes, which is not the default physics behavior.

godot 4.3 rc3

![Default](https://github.com/user-attachments/assets/5693d0d2-3972-42a1-af8a-37cf1bf11f26)

![Rapier2D](https://github.com/user-attachments/assets/813d1b0f-e239-45da-ab5b-02ceb56787de)

[r2d_bug.zip](https://github.com/user-attachments/files/16558974/r2d_bug.zip)

**Comment by Ughuuu** [comment]
Very strange, seems to work with circle shape for eg. Also works with square shape. Only the capsule shape seems to go a bit inside the shape.

**Comment by CodeLazier** [comment]
Yes, as long as the shape has any angle, the collision will be incorrect.
![2024-08-10_134538](https://github.com/user-attachments/assets/c6036f8a-63eb-412d-8960-b92152c46246)
![2024-08-10_134600](https://github.com/user-attachments/assets/bc5923cf-c032-41dc-b341-ba69f2af0303)
![2024-08-10_134619](https://github.com/user-attachments/assets/a3697418-c1fc-4b88-a275-55ea5bcb66c0)

**Comment by Ughuuu** [comment]
Y, dont worry, im just writting here what i see that doesnt work as i investigate. The thinking is that all will work once its fixed.

**Comment by Ughuuu** [comment]
Fixed now for scaling. Now only the rotation issue remains.

**Comment by Ughuuu** [comment]
Problem seems to be where I get aabb for character body I don't take into consideration rotation of shape.
